### PR TITLE
UX team based User selection view for an idp app.  

### PR DESCRIPTION
**Pull request by trooper2013**
Replaced out the previous UI screen with the UX team proposed design.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUITableViewCell.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ SFSDKUITableViewCell.h
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 6/05/18.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUITableViewCell.h]: `2018-present`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUITableViewCell.m [hunk 2]
```diff
@@ -0,0 +1,126 @@
+/*
+ SFSDKUITableViewCell.m
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 6/05/18.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUITableViewCell.m]: `2018-present`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUserSelectionTableViewController.m [hunk 3]
```diff
@@ -29,157 +29,279 @@
...
+    stack.spacing = 0;
     
-    NSMutableAttributedString *attributedText = [[NSMutableAttributedString alloc] initWithString:appName  attributes:@{ NSFontAttributeName : font, NSForegroundColorAttributeName : [UIColor redColor] }];
+    [stack setAxis:UILayoutConstraintAxisVertical];
+    [stack addArrangedSubview:headerView];
+    [stack addArrangedSubview:self.tableView];
+    [self.view addSubview:stack];
     
-    NSMutableAttributedString *attributedNormalText = [[NSMutableAttributedString alloc] initWithString:@" is requesting access to users credentials. Select a user from the list" attributes:@{NSFontAttributeName : normalFont , NSForegroundColorAttributeName : [UIColor grayColor] }];
+    if (@available(iOS 11, *)) {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUserSelectionTableViewController.m]: Can we make this such that it works for both `iOS 10` and `iOS 11` (basically fall back on the `iOS10` API)?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUserSelectionTableViewController.m [hunk 4]
```diff
@@ -29,157 +29,279 @@
...
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    SFSDKUITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:[SFSDKUITableViewCell reuseCellIdentifier] forIndexPath:indexPath];
     
+    SFUserAccount *userAccount = self.userData[indexPath.row];
+    cell.userName  = [NSString stringWithFormat:@"%@ %@",userAccount.idData.firstName, userAccount.idData.lastName];
+    cell.hostName = [NSString stringWithFormat:@"%@",userAccount.credentials.domain];
     
+    NSURL *url = userAccount.idData.profileUrl;
+    cell.imageURL = url;
+    //cell.profileImage = nil;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUserSelectionTableViewController.m]: Remove commented out code.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIColor+SFSDKIDP.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/*
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 6/05/18.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIColor+SFSDKIDP.h]: `2018-present`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIColor+SFSDKIDP.m [hunk 6]
```diff
@@ -0,0 +1,69 @@
+/*
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 6/05/18.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIColor+SFSDKIDP.m]: `2018-present`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIFont+SFSDKIDP.h [hunk 7]
```diff
@@ -0,0 +1,34 @@
+/*
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 6/05/18.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIFont+SFSDKIDP.h]: `2018-present`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIFont+SFSDKIDP.m [hunk 8]
```diff
@@ -0,0 +1,38 @@
+/*
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 6/05/18.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/UIFont+SFSDKIDP.m]: `2018-present`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m [hunk 9]
```diff
@@ -1425,6 +1427,23 @@ - (void)finalizeAuthCompletion:(SFSDKOAuthClient *)client
     }
 }
 
+- (void)retrieveUserPhotoIfNeeded:(SFUserAccount *)account{
+    if (account.idData.pictureUrl) {
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:account.idData.pictureUrl];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m]: Could we use `SFRestRequest` and `SFRestAPI` here instead of `NSMutableURLRequest` and `SFNetwork`?

File: shared/resources/SalesforceSDKResources.bundle/en.lproj/Localizable.strings [hunk 10]
```diff
@@ -105,3 +105,9 @@
 "devInfoTitle" = "Mobile SDK Dev Support";
 "devInfoBackButtonTitle" = "Back";
 "devInfoOKKey" = "OK";
+
+//IDP User Selection Screen
+"idpSelectUserLabel" = "Select your Account to log into"
```
Comment (bhariharan) [inline shared/resources/SalesforceSDKResources.bundle/en.lproj/Localizable.strings]: I think "Account" should be lowercase in this first sentence.

File: shared/resources/SalesforceSDKResources.bundle/en.lproj/Localizable.strings [hunk 11]
```diff
@@ -105,3 +105,9 @@
...
+"idpAddNewAccountLabel" = "Add new account"
+"idpSelectUserLabel" = "Select User"
```
Comment (bhariharan) [inline shared/resources/SalesforceSDKResources.bundle/en.lproj/Localizable.strings]: Lowercase "User".

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m (hunk 9 above)
Comment (bhariharan) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m]: We need `SFRestAPI` to take care of refresh too.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m [hunk 12]
```diff
@@ -38,8 +38,8 @@
 
 // Private constants
 
-static NSString * const kHttpHeaderAuthorization             = @"Authorization";
-static NSString * const kHttpAuthHeaderFormatString          = @"Bearer %@";
+NSString * const kHttpHeaderAuthorization             = @"Authorization";
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m]: Revert this change since we're switching to `SFRestAPI`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.h [hunk 13]
```diff
@@ -40,6 +40,16 @@ extern const NSTimeInterval kSFIdentityRequestDefaultTimeoutSeconds;
  */
 extern NSString * const kSFIdentityErrorDomain;
 
+/**
+ * The Authorization header value.
+ */
+extern NSString * const kHttpHeaderAuthorization;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.h]: Revert this change since we're switching to `SFRestAPI`. We don't need to `extern` this.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m (hunk 12 above)
Comment (trooper2013) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityCoordinator.m]: @bhariharan as we discussed offline, we will continue to use SFNetwork. No need for refresh since this call is made during auth completion.
